### DISCUSSION TITLE
Documentation or examples for chunking/splitting and gathering bam files ?

### DISCUSSION THREAD

**Discussion by colindaven** [nextflow-io/nextflow#3594] [Q&A]
Dear all,

Nextflow has great docs and patterns for splitting and gathering fasta, fastq, csv etc.

The only thing missing is the probably central file in bioinformatics (at least for alignment and variant calling), the bam file. Has anyone found any simple pipeline or example which does this easily ? Or does everyone just chunk fastqs and feed them through the pipeline in nextflow-world ?

Thanks,
Colin

**Comment by pditommaso** [comment]
So far, there's no native support. Tho we were thinking to include Biojava to implement this via a plugin. Tagging @pabloaledo

**Reply by abhi18av** [reply]
Mm, interesting point. I did some digging into this earlier and seems like the core concern of the `biojava-alignment` package is to basically _do_ the alignment using different algorithms https://github.com/biojava/biojava/tree/264ebe14ce88c28c432bc76bfa6c77696f79ba1f/biojava-alignment/src/main/java/org/biojava/nbio/alignment 

Maybe in the case for a BAM, it's worth considering the `htsjdk` library since it has a BAM file reader https://github.com/samtools/htsjdk/blob/master/src/main/java/htsjdk/samtools/BAMFileReader.java  and offers support for effective reads/writes for other file formats such as CRAM / VCF etc 

Thoughts?

**Reply by pditommaso** [reply]
Abinhav, you are right. Actually, I remembered @lindenb once made an implementation [for this](https://github.com/nextflow-io/nextflow/pull/1266). We could recover that code and turn it into a plugin.

**Reply by lindenb** [reply]
> Abinhav, you are right. Actually, I remembered @lindenb once made an implementation [for this](https://github.com/nextflow-io/nextflow/pull/1266). We could recover that code and turn it into a plugin.

at the end, I wrote a the samtools subcommand `samtools samples` . I now usually split the output of this command using a simple `Channel.spitCsv`

For more complex queries, I usually write some small htsjdk-based java codes in a `process` in a "Here Doc".
